### PR TITLE
Fix heap buffer overflow in selReadStream (detected by clang address sanitizer)

### PR DESCRIPTION
**Pull request by stweil**

### CODE REVIEW

**Comment by stweil** [comment]
The fix for tiffio breaks `pixcomp_reg`, therefore this needs separate examination. I'll remove that commit from the pull request.

File: src/sel1.c [hunk 1]
```diff
@@ -1431,17 +1431,14 @@ SEL     *sel;
 
     if (fgets(linebuf, sizeof(linebuf), fp) == NULL)
         return (SEL *)ERROR_PTR("error reading into linebuf", procName, NULL);
-    selname = stringNew(linebuf);
```
Comment (stweil) [inline src/sel1.c:1434]: The old code allocated 24 bytes for `selname`.

File: src/sel1.c [hunk 2]
```diff
@@ -1431,17 +1431,14 @@ SEL     *sel;
...
     sscanf(linebuf, "  ------  %200s  ------", selname);
```
Comment (stweil) [inline src/sel1.c:1434]: ... and here 201 bytes were written to `selname`.

**Comment by DanBloomberg** [comment]
I found and fixed the int32 --> uint32 issue before reading your PR.

Thank you for this, and for fixing the selname issue.

Seems I can't merge because of a build issue

**Comment by stweil** [comment]
The build issue on Ubuntu looks unrelated to this pull request. @egorpugin, can you help, please?

**Comment by egorpugin** [comment]
Ignore it. Error is unrelated to leptonica. It's on sw side, I'll check.

**Comment by egorpugin** [comment]
Should be ok now.

**Comment by DanBloomberg** [comment]
I just realized this was waiting.
Thank you.

**Pull request merged**